### PR TITLE
[Build Script Helper] Make the 'build' action use the CMake build flow

### PR DESCRIPTION
**Pull request by artemcm**
This allows us to decouple building swift-driver inside `swift`'s `build-script` from relying on SwiftPM.
This applies to both Darwin and Linux platforms.
The `Install` action is still different across Darwin and Linux, generalizing that will be followup work, once I resurrect my Linux environment. 

Resolves rdar://70060523

### CODE REVIEW

**Comment by artemcm** [comment]
@swift-ci please test

File: Utilities/build-script-helper.py [hunk 1]
```diff
@@ -334,42 +348,48 @@ def install_include_artifacts(args, toolchain_include_dir, src_include_dir, dst_
...
+  # Platform-specific targets for which we must build swift-driver
   if platform.system() == 'Darwin':
     targets = [x + '-apple-macos' + macos_deployment_target for x in macos_target_architectures]
   else:
     targets = [get_build_target(swiftc_path, args)]
 
+  swift_flags = []
+  if args.configuration == 'debug':
+    swift_flags.append('-Onone')
+    swift_flags.append('-DDEBUG=1')
```
Comment (DougGregor) [inline Utilities/build-script-helper.py:363]: I don't think the =1 is needed here

**Review by DougGregor — APPROVED** [review]
Thanks!

**Comment by artemcm** [comment]
Must be merged together with: https://github.com/apple/swift/pull/34233

**Comment by artemcm** [comment]
@swift-ci please test

**Pull request merged**